src/main/java/com/robertx22/mine_and_slash/mechanics/thresholds/SpendThresholdManager.java

```diff
@@ -66,21 +68,26 @@ public static void processSpend(ServerPlayer sp, EntityData unit, ResourceType t
             if (threshold <= 0f) continue;
 
             int procs = tracker.addAndConsumeForKey(key, type, loss, threshold);
-            // activity tracking omitted for compatibility
+            if (loss > 0f && procs == 0) {
+                unit.getSpendRuntime().markActivity(key, now);
+                unit.getSpendRuntime().markActive(type, key, spec);
+            }
             if (procs > 0) {
                 spec.onProc(sp, procs);
                 spec.startCooldown(unit, now);
                 if (spec.resetOnProc()) {
```

Comment:
To match names of other methods
```suggestion
                if (spec.dropProgressOnProc()) {
```